### ISSUE TITLE
Pinned rows duplicates

### ISSUE THREAD

**Issue by shadoworion** [ag-grid/ag-grid#5660]
**I'm submitting a ...**  (check one with "x")
```
[X] bug report => see 'Providing a Reproducible Scenario'
```

**Current behavior**
Since the last update, if you want to pin some rows, you see these rows as pinned and as regular, too - so I have duplicates.

**Expected behavior**
Before the last update, all pinned rows were not showing as regular rows - no duplicates.

* **AG Grid version:** 28.2.0

* **Language:** TypeScript, React, Vite

**Comment by hespy** [comment]
I don't know if it is related, so will comment here in case it helps with investigation and raise a separate issue, but seems coincidence as also related to pinning.

Since release of 28.2.0 when using pinned auto group columns, any changes to column definitions creates duplicate auto group columns.

**Comment by StephenCooper** [comment]
Can you help us produce a reproduction of the issue as our pinning example seems to be working as expected.

https://plnkr.co/edit/22eMgrQmz5HqsMHj?open=index.tsx

**Comment by shadoworion** [comment]
@StephenCooper 

No, sorry, maybe it is not clear. I am speaking about pinned rows (pinnedTopRowData), not pinned columns.

**Comment by StephenCooper** [comment]
Sorry, my mistake but could you still provide a reproduction so that we can investigate and see the code that you are using.

**Comment by shadoworion** [comment]
@StephenCooper 

https://plnkr.co/edit/C0Q8D03tveIy11Ki?preview

We have "Test6" two times - first as pinned row and second as a normal row
